src/main/clojure/cemerick/pomegranate/aether.clj

```diff
@@ -637,28 +626,68 @@ kwarg to the repository kwarg.
      (= (:classifier om)
         (:classifier som)))))
 
-(defn- find-managed-coord
-  "Given an entry from a coordinates vector, and a managed coordinates vector, find
-  the entry in the managed coordinates vector (if any) that matches the coordinate."
-  [coord managed-coords]
-  (first (filter #(coordinates-match? coord %) managed-coords)))
-
-(defn- add-version-from-managed-coords-if-missing
-  "Given a managed coordinates vector and an entry from a coordinates vector, check
-  to see if the coordinate specifies a version string, and if not, update it with
-  the version string from the managed coordinates (if it is defined)."
-  [managed-coords coord]
-  (if (nil? (second coord))
-    (add-version-from-managed-coord coord (find-managed-coord coord managed-coords))
-    coord))
+(defn- coord-map-key
+  "Creates the unique key for the given coordinate map.  This is
+   essentially a representation of the full maven coordinate except
+   the version."
+  [{:keys [project extension classifier]}]
+  [project extension classifier])
+
+(defn- canonical-id [id]
+  (apply symbol (distinct ((juxt group name) id))))
+
+(defn- conform-coord
+  "Returns a map describing the coordinate.  The full project ID is
+   under the :project key and the version, if specified and not nil,
+   under the :version key.  All other specified options appear with
+   the keys and values specified (including nil values)."
+  [[project & opts]]
+  (if project
+    (let [has-version? (odd? (count opts))
```

Comment:
@loomis nice... you might be interested in technomancy/leiningen#2223, which introduces some spec stuff for the full lein project map (and does so in a way that doesn't require Clojure 1.9 AFAIK).  It might be nice to try to consolidate the parts of the spec that are common - e.g. dependency vectors... and maybe pomegranate is a reasonable place to do that.  (Not suggesting it would need to be done as part of this PR or anything, just food for thought :) )